packages/@react-aria/autocomplete/src/useAutocomplete.ts

```diff
@@ -209,6 +219,7 @@ export function useAutocomplete<T>(props: AriaAutocompleteOptions<T>, state: Aut
         // close the dialog prematurely. Ideally that should be up to the discretion of the input element hence the check
         // for isPropagationStopped
         if (e.isDefaultPrevented()) {
+          setUpdated((last) => !last);
```

Comment:
adding a delay before calling this seems to fix searchfield but feels messy. Might be possible to bring back the top level controlled autocomplete input value state and rely on `filterText` instead of getting rid of value management at the Autocomplete level